src/ast/nodes/MetaProperty.ts

```diff
@@ -95,11 +96,15 @@ export default class MetaProperty extends NodeBase {
 			return;
 		}
 
-		const replacement =
-			pluginDriver.hookFirstSync('resolveImportMeta', [
-				metaProperty,
-				{ chunkId, format, moduleId }
-			]) || importMetaMechanisms[format]?.(metaProperty, { chunkId, snippets });
+		let replacement = pluginDriver.hookFirstSync('resolveImportMeta', [
+			metaProperty,
+			{ chunkId, format, moduleId }
+		]);
+		if (!replacement) {
+			replacement = importMetaMechanisms[format]?.(metaProperty, { chunkId, snippets });
+			module.accessedDocumentCurrentScript =
+				formatsMaybeAccessDocumentCurrentScript.includes(format) && replacement !== 'undefined';
```

Comment:
I see two problems with doing it this way:
- If there is more than one `import.meta.<something>` on a page and one is `undefined`, it will override this flag for others. So it might make sense to instead only change this flag if we would set it to `true` and otherwise leave it untouched so that it can never switch from `true` to `false`
- One should not leave state on `Module` during rendering. Otherwise, there is a real danger that when rendering several outputs, they will influence each other. Another way of doing it would be to extend the `RenderOptions` passed to `render` with a `Set<Module>` containing all modules that access document.currentScript. That would contain this state in the current render run. What do you think?

Reply:
Now that is even better. Nice work!